### PR TITLE
disable state_getStorage in prod

### LINKED ISSUE #1781: Disable sidechain `state_getStorage` in prod
### Context

Follow-up of DI: after we expose the DI port we should disallow the query of the storage for prod sidechain - for obvious reasons :)

Currently it's handled by the `state_getStorage` RPC call.

Please note it should be a compilation-time path deviation.

---
:heavy_check_mark: Please set appropriate **labels** and **assignees** if applicable.

### PR DESCRIPTION
**Pull request by kziemianek**
`state_getStorage` direct rpc call will be not available if `production` feature is enabled.

closes: #1781

### CODE REVIEW

File: tee-worker/cli/demo_shielding_unshielding.sh [hunk 1]
```diff
@@ -118,6 +138,7 @@ function assert_account_state()
 }
 
 echo "* Query on-chain enclave registry:"
+echo ${CLIENT}
```
Comment (kziemianek) [inline tee-worker/cli/demo_shielding_unshielding.sh]: I'll remove this line.

File: tee-worker/enclave-runtime/src/rpc/worker_api_direct.rs [hunk 2]
```diff
@@ -304,6 +258,56 @@ where
...
+						},
+						Err(e) => {
+							let error_msg = format!("load shard failure due to: {:?}", e);
+							return Ok(json!(compute_hex_encoded_return_error(error_msg.as_str())))
+						},
+					}
+				},
+				Err(_err) => Ok(json!(compute_hex_encoded_return_error("parse error"))),
+			}
+		});
```
Comment (BillyWooo) [inline tee-worker/enclave-runtime/src/rpc/worker_api_direct.rs:309]: Is it only switch the code position?

Comment (kziemianek) [reply tee-worker/enclave-runtime/src/rpc/worker_api_direct.rs:309]: Yes, it's moved inside condition `if cfg!(not(feature = "production"))` so this rpc method won't be present with `production` feature on.

**Review by Kailai-Wang — APPROVED** [review]
Looks good thanks!

**Pull request merged**